src/DownloadEmbeddings.targets

```diff
@@ -0,0 +1,54 @@
+<Project>
+    <PropertyGroup>
+        <LocalEmbeddingsModelCacheDir Condition="'$(LocalEmbeddingsModelCacheDir)' == ''">$(MSBuildThisFileDirectory)..\.cache\</LocalEmbeddingsModelCacheDir>
+
+        <!-- To use a custom model, developers must specify LocalEmbeddingsModelName and either LocalEmbeddingsModelUrl or LocalEmbeddingsModelPath -->
+        <!-- (likewise with LocalEmbeddingsVocabUrl or LocalEmbeddingsVocabPath) -->
+        <!-- Then at runtime, pass the model name to the LocalEmbeddings constructor, and it will use the corresponding model+vocab pair. -->
+        <LocalEmbeddingsModelName Condition="'$(LocalEmbeddingsModelName)' == ''">bge-micro-v2</LocalEmbeddingsModelName>
+
+        <!--
+            bge-micro-v2 comes from https://huggingface.co/TaylorAI/bge-micro-v2, which is forked to
+            https://huggingface.co/SmartComponents/bge-micro-v2 purely to ensure it remains available.
+            The license for the model remains MIT, but it's not redistributed in this repo or the NuGet
+            package anyway, since it's downloaded at build time.
+        -->
+        <LocalEmbeddingsModelUrl Condition="'$(LocalEmbeddingsModelUrl)' == ''">https://huggingface.co/SmartComponents/bge-micro-v2/resolve/72908b7/onnx/model_quantized.onnx</LocalEmbeddingsModelUrl>
+        <LocalEmbeddingsVocabUrl Condition="'$(LocalEmbeddingsVocabUrl)' == ''">https://huggingface.co/SmartComponents/bge-micro-v2/resolve/72908b7/vocab.txt</LocalEmbeddingsVocabUrl>
+
```

Comment:
Assume that we want to move to a new model, how is that going to work?
Existing indexes will need to use the existing model, no?

What is the forward compatability story?

Reply:
cc @maciejaszyk 
But in general not a problem, because this can be persisted in index storage, would prefer to implement that ability when this will be actually the case

Reply:
As Pawel mentioned, there is no need to handle it now, as the solution can be implemented via index storage.